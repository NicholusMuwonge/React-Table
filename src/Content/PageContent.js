import React, { Fragment, useEffect } from "react";
import axios from "axios";
import Table from "../Table/TableComponent";

const Content = () => {
  const [data, getfunction] = React.useState({});
  const [searchTerm, setSearchTerm] = React.useState("");

  const handleChange = event => {
    setSearchTerm(event.target.value);
  };

  var results = !searchTerm
    ? []
    : Object.values(data).map(m =>
        m.filter(person =>
          person.description
            .toLowerCase()
            .includes(searchTerm.trim().toLocaleLowerCase())
        )
      );

  const getdataUrl =
    "https://sievo-react-assignment.azurewebsites.net/api/data";

  useEffect(() => {
    axios
      .get(getdataUrl)
      .then(res => {
        const response = res.data;
        getfunction({ ...data, response });
      })
      .catch(err => console.log(err.res));
  }, []);

  return (
    <Fragment>
      <div className="container">
        <h3>Pricing Details</h3>
        <input
          type="search"
          placeholder="Search here.."
          onChange={handleChange}
          value={searchTerm}
        />
        {Object.values(data).length > 0 ? (
          <Table data={!searchTerm ? data : results} />
        ) : (
          <p className="loader">Loading ...</p>
        )}
      </div>
    </Fragment>
  );
};

export default Content;
