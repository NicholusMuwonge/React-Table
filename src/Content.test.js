import React from 'react'
import { unmountComponentAtNode } from "react-dom";

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from "enzyme";
import Content, { normaliseDate, normalizeCurrency } from "./PageContent";


configure({ adapter: new Adapter() });


it("should return date formated dd.mm.yyyy", () => {
  expect(normaliseDate("2013-12-28T00:00:00+00:00")).toBe("28.12.2013");
});

it("should return empty string if NULL is passed", () => {
  expect(normalizeCurrency("NULL")).toBe("");
});

describe("<Content /> with no props", () => {
  let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  var wrapper = shallow(<Content data={[]}/>);

  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  
  var table = wrapper.find('table');
  it("should have one table element", () => {
    expect(table).toHaveLength(1);
  });

  it("should have one table element", () => {
    var table = wrapper.find('table');
    const thead = table.find('thead');
    const th = thead.find('th');
    expect(th).toEqual({});
    expect(th).toHaveLength(8);
  });



});
