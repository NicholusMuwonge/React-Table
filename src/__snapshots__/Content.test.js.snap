// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> with no props should match the snapshot 1`] = `"<div class=\\"container\\"><input type=\\"search\\" placeholder=\\"Search here..\\" value=\\"\\"/><img src=\\"sort.svg\\" alt=\\"\\" style=\\"height:1.4rem;margin-left:10px;cursor:pointer\\"/><table class=\\"table table-responsive-sm\\"><thead><tr><th scope=\\"col\\">Project</th><th scope=\\"col\\">Description </th><th scope=\\"col\\">Start Date</th><th scope=\\"col\\">Category</th><th scope=\\"col\\">Responsible</th><th scope=\\"col\\">Savings Amount</th><th scope=\\"col\\">Currency</th><th scope=\\"col\\">Complexity</th></tr></thead><tbody></tbody></table></div>"`;
